crates/fluvio-socket/src/versioned.rs

```diff
@@ -248,12 +248,21 @@ impl VersionedSerialSocket {
         self.socket.clone()
     }
 
+    /// Check if inner socket is stale
+    pub fn is_stale(&self) -> bool {
+        self.socket.is_stale()
+    }
+
     /// send and wait for reply serially
     #[instrument(level = "trace", skip(self, request))]
     pub async fn send_receive<R>(&self, request: R) -> Result<R::Response, SocketError>
     where
         R: Request + Send + Sync,
     {
+        if self.is_stale() {
```

Comment:
let's make it reusable instead of repeating this.  So something like this:

```
/// check to make sure connection is ok
fn check_liveness() -> Result<(),SocketErrro> {
         if self.is_stale() {
        ....
}
```

Then this line just could be: ```
self.check_liveness()?
```